Contest_API.md

```diff
@@ -773,12 +785,22 @@ JSON elements of group objects:
 | icpc\_id | string  | no        | yes       | CCS        | external identifier from ICPC CMS                                        |
 | name     | string  | yes       | no        | CCS        | name of the group                                                        |
 | type     | string  | no        | yes       | CCS        | type of this group                                                       |
-| hidden   | boolean | no        | yes       | CCS        | if group should be hidden from scoreboard. Defaults to false if missing. |
+| hidden   | boolean | no        | yes       | CCS        | if the group should be excluded from the [scoreboard](#scoreboard). Defaults to false if missing. |
 
 #### Access restrictions at WF
 
 No access restrictions apply to a GET on this endpoint.
 
+#### Known group types
+
+The list below contains standardized identifiers for known group
+types. These identifiers should be used when the purpose
+of a group matches.
+
+| Type  | Name                | Description
+| ----- | ------------------- | ---------------------------------
+| site  | Site                | A physical location where teams are competing, e.g. the "Hawaii site"
```

Comment:
Warning: Not actually a comment on this PR.

Wait, what? Is the ending pipe optional? They are always annoying. A world of possibilities opens up...

Reply:
*Thank you* for ending my day with a laugh. Last night when I copied this from one of the other 'known groups' sections I was trying to figure out why the author took the time to put in all the end pipes. lol :)

Reply:
The way this table is written makes it sound like it's something that we explicitly refer to from other parts of the spec, but that's not the case, right? Also, the columns `Type` and `Name` make it unclear to me whether the site type should be put in as `site` or `Site`. I guess the former in lower case, so do we need the `Name` column?

Also, we should add at least `division` here, and maybe others we know of?

Reply:
> The way this table is written makes it sound like it's something that we explicitly refer to from other parts of the spec, but that's not the case, right?

Correct. It's mostly just a hint for clients.

> Also, the columns `Type` and `Name` make it unclear to me whether the site type should be put in as `site` or `Site`. I guess the former in lower case, so do we need the `Name` column?

Yeah, I waffled on this before. Name column removed now.

> Also, we should add at least `division` here, and maybe others we know of?

On the call I had to argue that site should be listed because it has a useful meaning that would change defaults or behaviour for many tools (e.g. balloon printer, scoreboard for site, awards for site, etc). I'm not sure you can say the same for division, and while I don't think there's any harm in including I'd rather leave out to start and confirm at the next meeting.

Reply:
> On the call I had to argue that site should be listed because it has a useful meaning that would change defaults or behaviour for many tools (e.g. balloon printer, scoreboard for site, awards for site, etc). 

You didn't have to argue *that* hard :).

> I'm not sure you can say the same for division, and while I don't think there's any harm in including I'd rather leave out to start and confirm at the next meeting.

Not only that, I don't think we actually have any known instances of division being used (or rather, known instances where division would have been used had the system followed the spec we are creating). Remember that the PacNW or SEUSA style divisions would **not** use a group of type division, rather they would be separate contests.

Well... actually, the NADC divisions would fit perfectly, but that is the first use, and not how we have typically thought about divisions. So, I'm not against adding division at some point, but we should at least have a good feeling for what that entails. If the only semantic meaning is "some sort of grouping", i.e. if it adds no additional semantic value on "being a group" and possibly "different types are different", then IMO we shouldn't add it.

Other kinds of groups that have been used in the past are "countries" (i.e. "Sweden", "Finland", "Norway", a.s.o.) or "subcontests" (i.e. "NCPC", "BAPC", "UKIEPC", a.s.o.). I don't think the former is a good choice since the concept of "country" has a lot of baggage, I would prefer to generalize it to something like "geographic area". The latter might be useful though?

Reply:
'Lightly convince', maybe? ;-)

I do agree with the rest. Site has a clear meaning, is being used, and can improve client behaviour, we should generally have similar reasons to add others.